File: DR_system/Demos/VB/Barcode/Form1.vb
Imports System.ComponentModel
Imports System.Text
Imports System.Drawing.Drawing2D

Imports Spire.Barcode

Namespace Barcode
	Partial Public Class Form1
		Inherits Form
		Public settings As BarcodeSettings
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnCreate_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCreate.Click
			'set the configuration of barcode
			settings = New BarcodeSettings()
			Dim data As String = "12345"
			Dim type As String = "Code128"
			If Me.textBoxText.Text IsNot Nothing AndAlso Me.textBoxText.Text.Length > 0 Then
				data = Me.textBoxText.Text
			End If

            settings.Data2D = data
            settings.Data = Me.textBoxText.Text

			If comboBoxType.SelectedItem IsNot Nothing Then
				type = comboBoxType.SelectedItem.ToString()
			End If
			settings.Type = CType(System.Enum.Parse(GetType(BarCodeType), type), BarCodeType)

            If Me.checkBoxBorder.Checked Then
                If comboBoxText.SelectedItem IsNot Nothing Then
                    settings.HasBorder = True
                    settings.BorderDashStyle = CType(System.Enum.Parse(GetType(DashStyle), comboBoxText.SelectedItem.ToString()), DashStyle)
                End If
            End If

            Dim fontSize As Short = 8
            Dim font As String = "SimSun"
            If Me.comboBoxFont.SelectedItem IsNot Nothing Then
                font = Me.comboBoxFont.SelectedItem.ToString()
            End If

            If Me.textBoxSize.Text IsNot Nothing AndAlso Me.textBoxSize.Text.Length > 0 AndAlso Int16.TryParse(Me.textBoxSize.Text, fontSize) Then
                If font IsNot Nothing AndAlso font.Length > 0 Then
                    settings.TextFont = New Font(font, fontSize, FontStyle.Bold)
                End If
            End If

            Dim barHeight As Short = 15
            If Me.textBoxHeight.Text IsNot Nothing AndAlso Me.textBoxHeight.Text.Length > 0 AndAlso Int16.TryParse(Me.textBoxHeight.Text, barHeight) Then
                settings.BarHeight = barHeight
            End If
            If Me.checkBoxText.Checked Then
                settings.ShowText = True
            Else
                settings.ShowText = False
            End If

            If Me.checkBoxSum.Checked Then
                settings.ShowCheckSumChar = True
            Else
                settings.ShowCheckSumChar = False
            End If

            If Me.comboBoxColor.SelectedItem IsNot Nothing Then
                Dim foreColor As String = Me.comboBoxColor.SelectedItem.ToString()
                settings.ForeColor = Color.FromName(foreColor)
            End If

            'generate the barcode use the settings
            Dim generator As New BarCodeGenerator(settings)
            Dim barcode As Image = generator.GenerateImage()

            'save the barcode as an image
            barcode.Save("..\..\..\..\Data\barcode.png")

            'launch the generated barcode image
            Process.Start("..\..\..\..\Data\barcode.png")
		End Sub

		Private Sub btnScan_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnScan.Click
			'scan the barcode
            Dim datas() As String = BarcodeScanner.Scan("..\..\..\..\Data\barcode.png")

			'show the scan result
			Me.textBox1.Text = datas(0)
		End Sub
	End Class
End Namespace

File: DR_system/Demos/CS/Barcode/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Drawing.Drawing2D;

using Spire.Barcode;

namespace Barcode
{
    public partial class Form1 : Form
    {
        public BarcodeSettings settings;
        public Form1()
        {
            InitializeComponent();
        }

        private void btnCreate_Click(object sender, EventArgs e)
        {
            //set the configuration of barcode
            settings = new BarcodeSettings();
            string data = "12345";
            string type = "Code128";
            if (this.textBoxText.Text != null && this.textBoxText.Text.Length > 0)
            {
                data = this.textBoxText.Text;
            }

            settings.Data2D = data;
            settings.Data = this.textBoxText.Text;

            if (comboBoxType.SelectedItem != null)
            {
                type = comboBoxType.SelectedItem.ToString();
            }
            settings.Type = (BarCodeType)Enum.Parse(typeof(BarCodeType), type);

            if (this.checkBoxBorder.Checked)
            {
                if (comboBoxText.SelectedItem != null)
                {
                    settings.HasBorder = true;
                    settings.BorderDashStyle = (DashStyle)Enum.Parse(typeof(DashStyle), comboBoxText.SelectedItem.ToString());
                }
            }
           
            short fontSize=8;
            string font = "SimSun";
            if (this.comboBoxFont.SelectedItem != null)
            {
                font=this.comboBoxFont.SelectedItem.ToString();
            }

            if (this.textBoxSize.Text != null && this.textBoxSize.Text.Length > 0 && Int16.TryParse(this.textBoxSize.Text, out fontSize))
            {
                if (font != null && font.Length > 0)
                {
                    settings.TextFont = new System.Drawing.Font(font, fontSize, FontStyle.Bold);
                }
            }

            short barHeight=15;
            if (this.textBoxHeight.Text != null && this.textBoxHeight.Text.Length > 0 && Int16.TryParse(this.textBoxHeight.Text, out barHeight))
            {
                settings.BarHeight = barHeight;
            }
            if (this.checkBoxText.Checked)
            {
                settings.ShowText = true;
            }
            else
            {
                settings.ShowText = false;
            }

            if (this.checkBoxSum.Checked)
            {
                settings.ShowCheckSumChar = true;
            }
            else
            {
                settings.ShowCheckSumChar = false;
            }

            if (this.comboBoxColor.SelectedItem != null)
            {
                string foreColor = this.comboBoxColor.SelectedItem.ToString();
                settings.ForeColor = Color.FromName(foreColor);
            }

            //generate the barcode use the settings
            BarCodeGenerator generator = new BarCodeGenerator(settings);
            Image barcode=generator.GenerateImage();

            //save the barcode as an image
            barcode.Save(@"..\..\..\..\Data\barcode.png");

            //launch the generated barcode image
            System.Diagnostics.Process.Start(@"..\..\..\..\Data\barcode.png");
        }

        private void btnScan_Click(object sender, EventArgs e)
        {
            //scan the barcode
            string[] datas = BarcodeScanner.Scan(@"..\..\..\..\Data\barcode.png");

            //show the scan result
            this.textBox1.Text = datas[0];
        }
    }
}
